apps/condo/domains/notification/adapters/redStore/redStoreNotificationSender.js

```diff
@@ -0,0 +1,69 @@
+const { get } = require('lodash')
+
+
+const { fetch } = require('@open-condo/keystone/fetch')
+
+const { APS_RESPONSE_STATUS_SUCCESS } = require('../apple/constants')
+
+
+class RedStoreNotificationSender {
+    _projectId = null
+    _serviceToken = null
+    _url = null
+
+    /**
+     * @param config
+     */
+    constructor (config) {
+        this._projectId = config['project_id']
+        this._serviceToken = config['service_token']
+        this._url = config['url']
+        this.sendPush = this.sendPush.bind(this)
+    }
+
+    async sendPush (token, notification, data) {
+        return await fetch(this._url, {
+            body: JSON.stringify({
+                message: {
+                    token,
+                    notification,
+                    data,
+                },
+            }),
+            method: 'POST',
+            headers: {
+                'Content-Type': 'application/json',
+                'Authorization': `Bearer ${this._serviceToken}`,
+            },
+        })        
+    }
+
+    async sendAll (notifications) {
+        const responses = []
+        let successCount = 0, failureCount = 0
+
+        for (let idx = 0; idx < notifications.length; idx++) {
+            const { token, data: pushData = {}, notification  = {}, type, appId } = notifications[idx]
+
+            const response = await this.sendPush(token, notification, pushData)
+            responses.push(await response.json())
```

Comment:
Please recheck - how it going to work in case if network issues occurred? Isn't it throws any error? maybe wrap it in try/catch? but probably such cases are covered inside fetch impl, please check that.

Reply:
inside fetch request already wrapped in try/catch. But after getting error we throw another error. So i add check for this case
148 line
https://github.com/open-condo-software/condo/blob/main/packages/keystone/fetch.js